packages/file-storage-common/src/azure-block-download.ts

```diff
@@ -27,10 +28,25 @@ export class AzureBlockDownload {
     dataEncryptionKey: string | null,
     strategy: CipherStrategy | null,
     encryptionArtifact: any,
-    range: string | null
+    range: string | null,
+    onCancel?: any
   ) {
     if (range) {
-      const block = await BlobStorage.getBlock(this.url, range);
+      let block: any;
+      if (onCancel) {
+        const source = axios.CancelToken.source();
+        const cancelToken = source.token;
+        block = await Promise.race([BlobStorage.getBlock(this.url, range, cancelToken), onCancel]);
+        if (block === 'cancel') {
```

Comment:
I think this needs to be pre-bound. It's not documented that you need to call cancel with the exact string of `'cancel'` 

i.e. in your `fileDownloadBrowserWithCancel` method it should look something like this:

```
  let cancel;
  const promise = new Promise((resolve, reject) => (cancel = () => resolve('cancel')));
```

Then the user can just call `cancel()` without having to pass in the magic string.